File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		HashSet<String> gomgom = new HashSet<>();
		int count = 0;
		int N = Integer.parseInt(br.readLine());
		for(int i = 0; i < N; i++){
			String input = br.readLine();
			if(input.equals("ENTER")){
				gomgom.clear();
			}else if(!gomgom.contains(input)){
				gomgom.add(input);
				count++;
			}
		}
		br.close();
		System.out.println(count);
	}
}

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
import sys
input = sys.stdin.readline

N = int(input())
count = 0
gomgom = set()
for i in range(N):
    inputChat = input().strip()
    if inputChat == "ENTER":
        gomgom.clear()
    elif inputChat not in gomgom:
        gomgom.add(inputChat)
        count += 1

print(count)